Modifiying the hyperlinks with relative paths

AmarnatReddy opened the pull request:
# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs

# Checklist:

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test

rakeshgm commented on utility/utils.py:
```diff
@@ -382,13 +382,7 @@ def configure_logger(test_name, run_dir, level=logging.INFO):
-        url_base=url_base, run_dir=run_dir_name, log_name=full_log_name
-    )
-
+    log_url = full_log_name
```
you have removed `url_base`, any reason wjy ?

AmarnatReddy replied:
Hi Rakesh,
url_base was used for formulating the absolute path of the log_file.
As I was giving a relative path for log files, Url_base was becoming an unused variable so I have removed it.
I have removed the whole block of code which is appending the url_base,dir_name and full_log_name
As we click on the log file link from resuts.html page (URL: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1614674135116/result.html) with only log_file_name in href. it gets appended to http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1614674135116/ + <log_file_name> .
Log_url gets formulated from the **current page url/<logfile_name>**.

Please find the sample test executed after changing this code:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1614925690996/

psathyan commented:
Please add PR description and kindly squash your commits

AmarnatReddy commented:
This is to fix https://github.com/red-hat-storage/cephci/issues/294 
We have removed the hardcoding for base_url for result.html page.
we have replaced href with only log file name.
With this change, when you try to navigate to the respective log from result.html log file name gets appended to the base url from the current page which we are broswing 
ex : if we have url as https://<base_url>/result.html. Onclick of link it formulates url to https://<base_url>/<log_file_name>

sunilkumarn417 commented:
IMO, At any time we need run-id as relative path for redirection href link and filename would solve problem either for local log path or shared log path `/ceph/cephci-jenkins/`. 
```
RUN-ID(#1)
	- results.html			# Since log files and results.html is under RUN-ID directory, 
	- task-1.log			# we need just the log filenames here ex., task-1.log 
	- task-2.log
	- startup.log
```

If we have log file name, I think we dont need to split abs_path
``` 
<td><a style="color: green;" href="{{ <file-name: ex., task-1.log> }}">{{ test.status }}</a></td>


<a style="color: green;" href="setup_install_pre-requisistes_0.log">Pass</a>
```

AmarnatReddy commented:
> IMO, At any time we need run-id as relative path for redirection href link, since log files and results.html on same run-id directory, the filename would solve problem either for local log path or shared log path `/ceph/cephci-jenkins/`.
> 
> ```
> RUN-ID(#1)
> 	- results.html			# Since log files and results.html is under RUN-ID directory, 
> 	- task-1.log			# we need just the log filenames here ex., task-1.log 
> 	- task-2.log
> 	- startup.log
> ```
> 
> If we have log file name, I think we dont need to split abs_path
> 
> ```
> <td><a style="color: green;" href="{{ <file-name: ex., task-1.log> }}">{{ test.status }}</a></td>
> 
> 
> <a style="color: green;" href="setup_install_pre-requisistes_0.log">Pass</a>
> ```

AmarnatReddy commented:
> IMO, At any time we need run-id as relative path for redirection href link, since log files and results.html on same run-id directory, the filename would solve problem either for local log path or shared log path `/ceph/cephci-jenkins/`.
> 
> ```
> RUN-ID(#1)
> 	- results.html			# Since log files and results.html is under RUN-ID directory, 
> 	- task-1.log			# we need just the log filenames here ex., task-1.log 
> 	- task-2.log
> 	- startup.log
> ```
> 
> If we have log file name, I think we dont need to split abs_path
> 
> ```
> <td><a style="color: green;" href="{{ <file-name: ex., task-1.log> }}">{{ test.status }}</a></td>
> 
> 
> <a style="color: green;" href="setup_install_pre-requisistes_0.log">Pass</a>
> ```

Hi Sunil,

With only file name it was working fine when we navigate from result.html file.
But when we try to navigate from email notification it was not able to append the base_url
so I have used split in case.

AmarnatReddy commented on utility/utils.py:
```diff
@@ -726,9 +726,18 @@ def create_html_file(test_result) -> str:
         info=info,
     )
-
+    for tc in test_results:
```
Hi Satya,

I have moved the link modification to template file.

AmarnatReddy commented:
Please find the log files generated after executing with the above PR 

http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1615185231365/

psathyan commented:
There is an empty file as part of your commit. Please remove it

psathyan commented on utility/utils.py:
```diff
@@ -382,8 +382,11 @@ def configure_logger(test_name, run_dir, level=logging.INFO):
     _root.addHandler(_handler)
-
-    url_base = "http://magna002.ceph.redhat.com/cephci-jenkins"
+    url_base = (
```
This looks good, thanks for fixing the links.

Would it make sense to have a variable having the value `http://magna002.ceph.redhat.com/cephci-jenkins`? For example `file_server_url` or `magna_url`.

psathyan commented on utility/utils.py:
```diff
@@ -702,9 +705,10 @@ def create_html_file(test_result) -> str:
-
-    log_link = "http://magna002.ceph.redhat.com/cephci-jenkins/{run}/".format(
-        run=run_name
+    log_link = (
```
I believe we can attempt to do string concatenation.

Maybe also add a comment on why we are looking for '/ceph/cephci-jenins' in the `run_dir` variable.

psathyan commented on utility/utils.py:
```diff
@@ -725,10 +729,25 @@ def create_html_file(test_result) -> str:
+        use_abs_log_link=True,
+    )
+
+    # Rendering the result.html with only relative paths for log file links
```
Can we reword? For example
```
Result.html file is stored in the folder containing the log files. Moving to relative path facilitate the copying of the
files to a different location without breaking the hyperlinks in result.html. 
```
